src/ParallelAlgorithms/Interpolation/Events/Interpolate.hpp

```diff
@@ -89,8 +94,21 @@ class Interpolate<VolumeDim, InterpolationTargetTag,
     static_assert(
         std::is_same_v<typename Metavariables::interpolator_source_vars,
                        tmpl::list<InterpolatorSourceVarTags...>>);
-    interpolate<InterpolationTargetTag>(temporal_id, mesh, cache, array_index,
-                                        interpolator_source_vars...);
+    if constexpr (Parallel::is_dg_element_collection_v<ParallelComponent>) {
+      const auto core_id = static_cast<int>(
+          Parallel::local_synchronous_action<
```

Comment:
Not a change request, but will this impact performance? This runs on every element any time an interpolation is triggered which means it'll keep locking the node group box to get the element collection tag.

Reply:
local synch actions don't lock

Reply:
Oh right it just passes the node lock along, and then the action can decide to actually lock it. Forgot about that